Content.Server/Atmos/EntitySystems/GasMinerSystem.cs

```diff
@@ -27,14 +27,44 @@ private void OnMinerUpdated(Entity<GasMinerComponent> ent, ref AtmosDeviceUpdate
 
             if (!GetValidEnvironment(ent, out var environment))
             {
-                miner.Idle = true;
+                if (miner.Enabled)
+                {
+                    miner.Enabled = false;
+                    Dirty(ent);
+                }
                 return;
             }
+            else
+            {
+                if (!miner.Enabled)
+                {
+                    miner.Enabled = true;
+                    Dirty(ent);
+                }
+            }
 
             // SpawnAmount is declared in mol/s so to get the amount of gas we hope to mine, we have to multiply this by
             // how long we have been waiting to spawn it and further cap the number according to the miner's state.
             var toSpawn = CapSpawnAmount(ent, miner.SpawnAmount * args.dt, environment);
-            miner.Idle = toSpawn == 0;
+
+            if (toSpawn == 0)
+            {
+                if (!miner.Idle)
+                {
+                    miner.Idle = true;
+                    Dirty(ent);
+                }
+            }
+            else
+            {
+                if (miner.Idle)
+                {
+                    miner.Idle = false;
+                    Dirty(ent);
+                }
+            }
```

Comment:
how aggressive should I be about making sure `Dirty` is called only when I'm sure the state is actually changing

Reply:
Pretty aggressive.